### ISSUE TITLE
Change gene fetching architecture

### ISSUE THREAD

**Issue by JobLeonard** [linnarsson-lab/loom-viewer#109]
We have a very specific bug: if we colour by a gene in scatterplot, then filter out a specific value (say, the zeros), then refresh, the app will send infinite incorrect row requests. The reason it does this has to do with the gene not being fetched yet at the time of filtering.

I can write a hack that works around this, but I think is solved best by changing how we fetch genes, because this will also make it easier to change functionality in the future and prevent future bugs.

In simple terms, right now fetching genes is done _explicitly_, but what would work better if it happens _implicitly_ whenever a gene is selected in some view's state and the gene is not present yet. This will remove the need to think about fetching data, and on the view side we only have to check if a gene/attribute is not null (which we're doing already anyway).
